src/main/java/org/mapstruct/intellij/codeinsight/references/MapstructMappingQualifiedByNameReference.java

```diff
@@ -0,0 +1,173 @@
+
+import static com.intellij.codeInsight.AnnotationUtil.findAnnotation;
+import static com.intellij.codeInsight.AnnotationUtil.getStringAttributeValue;
+import static org.mapstruct.intellij.util.MapstructUtil.asLookupWithRepresentableText;
+
+/**
+ * Reference for {@link org.mapstruct.Mapping#qualifiedByName()}.
+ *
+ * @author Oliver Erhart
+ */
+class MapstructMappingQualifiedByNameReference extends MapstructBaseReference {
+
+    /**
+     * Create a new {@link MapstructMappingQualifiedByNameReference} with the provided parameters
+     *
+     * @param element the element that the reference belongs to
+     * @param previousReference the previous reference if there is one (in nested properties for example)
+     * @param rangeInElement the range that the reference represent in the {@code element}
+     * @param value the matched value (useful when {@code rangeInElement} is empty)
+     */
+    private MapstructMappingQualifiedByNameReference(PsiElement element,
+                                                     MapstructMappingQualifiedByNameReference previousReference,
+                                                     TextRange rangeInElement, String value) {
+        super( element, previousReference, rangeInElement, value );
+    }
+
+    @Override
+    PsiElement resolveInternal(@NotNull String value, @NotNull PsiType psiType) {
+        return null; // not needed
+    }
+
+    @Override
+    PsiElement resolveInternal(@NotNull String value, @NotNull PsiMethod mappingMethod) {
+
+        List<PsiMethod> allMethodsFromThisAndReferencedMappers = findAllMethodsFromThisAndReferencedMappers(
+            mappingMethod );
+
+        return allMethodsFromThisAndReferencedMappers.stream()
+            .filter( MapstructUtil::isNamedMethod )
+            .filter( this::methodHasReturnType )
+            .filter( m -> findAnnotation( m, true, MapstructUtil.NAMED_ANNOTATION_FQN ) != null )
+            .filter( a -> Objects.equals( getNamedValue( a ), value ) )
+            .findAny()
+            .orElse( null );
+    }
+
+    @Nullable
+    private String getNamedValue(PsiMethod method) {
+
+        PsiAnnotation annotation = findAnnotation( method, true, MapstructUtil.NAMED_ANNOTATION_FQN );
+
+        if ( annotation == null ) {
+            return null;
+        }
+
+        return getStringAttributeValue( annotation, "value" );
+    }
+
+    @NotNull
+    @Override
+    Object[] getVariantsInternal(@NotNull PsiType psiType) {
+        return LookupElement.EMPTY_ARRAY; // not needed
+    }
+
+    @NotNull
+    @Override
+    Object[] getVariantsInternal(@NotNull PsiMethod mappingMethod) {
+
+        List<PsiMethod> allMethodsFromThisAndReferencedMappers = findAllMethodsFromThisAndReferencedMappers(
+            mappingMethod );
+
+        return allMethodsFromThisAndReferencedMappers.stream()
+            .filter( MapstructUtil::isNamedMethod )
+            .filter( this::methodHasReturnType )
```

Comment:
Let's move these checks in the `findAllMethodsFromThisAndReferencedMappers`.  That one is already partially doing this for the named methods in used mappers.